lib/core/pktmbuf/pktmbuf.h

```diff
@@ -1153,6 +1194,101 @@ const char *cne_get_tx_ol_flag_name(uint64_t mask);
  */
 int cne_get_tx_ol_flag_list(uint64_t mask, char *buf, size_t buflen);
 
+/**
+ * Return the metadata buffer address
+ *
+ * @param m
+ *   The pktmbuf_t structure pointer
+ * @return
+ *   NULL on error or pointer to metadata buffer
+ */
+static inline void *
+pktmbuf_metadata(const pktmbuf_t *m)
+{
+    pktmbuf_info_t *p;
+
+    if (!m)
+        return NULL;
+
+    if (((p = m->pooldata) != NULL) && p->metadata)
+        return CNE_PTR_ADD(p->metadata, (m->meta_index * p->metadata_bufsz));
+    else
+        return CNE_PTR_ADD(m, sizeof(pktmbuf_t)); /* default to metadata in pktmbuf headroom */
+}
+
+/**
+ * Return the size of the metadata buffer if external or headroom size if internal buffer.
+ *
+ * @param m
+ *   The pktmbuf_t structure pointer
+ * @return
+ *   -1 if pktmbuf pointer is NULL or size of metadata buffer or headroom size if internal.
+ */
+static inline uint32_t
+pktmbuf_metadata_bufsz(const pktmbuf_t *m)
+{
+
+    pktmbuf_info_t *p;
+
+    if (!m)
+        return -1;
+
+    if (((p = m->pooldata) != NULL) && p->metadata)
+        return p->metadata_bufsz;
+    else
+        return (uint32_t)pktmbuf_headroom(m);
+}
+
+/**
+ * Construct a pktmbuf_pool_cfg_t structure with the given parameters.
+ *
+ * @param c
+ *   Pointer to the pktmbuf_pool_cfg_t structure
+ * @param addr
+ *   Address of the pktmbuf buffers, must not be NULL
+ * @param bufcnt
+ *   Number of buffers in the pktmbuf pool, must not be zero
```

Comment:
The pktmbuf_pool_create() function (prior to this change) let users pass a value of 0 for bufcnt and/or bufsz to indicate a default value should be chosen. Do you think we should keep that? I like it, since its less to configure...

Reply:
Changed the text to allow them to zero and I do not see if the code needs to change now after the above change.